From 41d5cf78bdcd4f8bbbcbb85cc3c0bc3e5ef05dbb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Tue, 9 Jun 2026 09:26:38 +0300
Subject: [PATCH] av1parser: Fix bytes/bits confusion when parsing tile data
 size

Which would then later lead to an assertion.

Patch provided by Junyi Liu, who also reported this.

Fixes https://gitlab.freedesktop.org/gstreamer/gstreamer/-/work_items/5103

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/11803>
---
 subprojects/gst-plugins-bad/gst-libs/gst/codecparsers/gstav1parser.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/subprojects/gst-plugins-bad/gst-libs/gst/codecparsers/gstav1parser.c b/subprojects/gst-plugins-bad/gst-libs/gst/codecparsers/gstav1parser.c
index e2caf91..cf2d860 100644
--- a/subprojects/gst-plugins-bad/gst-libs/gst/codecparsers/gstav1parser.c
+++ b/subprojects/gst-plugins-bad/gst-libs/gst/codecparsers/gstav1parser.c
@@ -4339,6 +4339,8 @@ gst_av1_parser_parse_tile_list_obu (GstAV1Parser * parser,
   }
 
   for (tile = 0; tile <= tile_list->tile_count_minus_1; tile++) {
+    guint32 tile_data_size;
+
     if (AV1_REMAINING_BITS (br) < 8 + 8 + 8 + 16) {
       retval = GST_AV1_PARSER_NO_MORE_DATA;
       goto error;
@@ -4353,8 +4355,8 @@ gst_av1_parser_parse_tile_list_obu (GstAV1Parser * parser,
     tile_list->entry[tile].coded_tile_data =
         obu->data + gst_bit_reader_get_pos (br) / 8;
     /* skip the coded_tile_data */
-    if (!gst_bit_reader_skip (br,
-            tile_list->entry[tile].tile_data_size_minus_1 + 1)) {
+    tile_data_size = tile_list->entry[tile].tile_data_size_minus_1 + 1;
+    if (!gst_bit_reader_skip (br, tile_data_size * 8)) {
       retval = GST_AV1_PARSER_NO_MORE_DATA;
       goto error;
     }
-- 
2.52.0

